### PR TITLE
updates to export image code

### PR DESCRIPTION
**Pull request by mihirsamdarshi**

### CODE REVIEW

**Comment by mihirsamdarshi** [comment]
@dondi a couple questions. The first that I have is to resolve the fact that the save svg to png package returns two promises, both PNG files. Is there any way that I can create a function that waits until both have been returned, to determine which one to save? And the second is that my save svg to svg function doesn't seem to save a file. Can you help me figure out why that is happening? When I click on the link to the blob generated, it says "Cannot GET _link to file_"

**Comment by dondi** [comment]
@mihirsamdarshi The first one is quick so I’ll answer that here: you want [Promise.all](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all) for that.

For the second question I’ll need to look at the code…if I don’t get to it before the meeting, we can take a look at that time.

File: web-client/public/js/graph.js [hunk 1]
```diff
@@ -654,27 +653,27 @@ export var drawGraph = function (network) {
...
+                return d.value.toPrecision(4);
+            })
+            .each(function (d) {
+                d.weightElement = d3.select(this);
+            });
 
     }
 
-  /* Big thanks to the following for the smart edges
+    /* Big thanks to the following for the smart edges
```
Comment (dondi) [inline web-client/public/js/graph.js]: I see a lot of re-indenting in the latest changes to this PR and most of them look right—but this one doesn’t. Note where the asterisks are in the succeeding lines.
```suggestion
  /* Big thanks to the following for the smart edges
```

File: web-client/public/js/graph.js [hunk 2]
```diff
@@ -1332,7 +1333,7 @@ export var drawGraph = function (network) {
                 return "translate(" + d.x + "," + d.y + ")";
             });
 
-      /* Allows for looping edges.
+            /* Allows for looping edges.
```
Comment (dondi) [inline web-client/public/js/graph.js]: The reindenting on this comment looks off as well.
```suggestion
      /* Allows for looping edges.
```

File: web-client/public/js/setup-handlers.js [hunk 3]
```diff
@@ -75,9 +91,29 @@ export const setupHandlers = grnState => {
         updateApp(grnState);
     };
 
-    $(GRID_LAYOUT_BUTTON).click(() => setGraphLayout(grnState.graphLayout === FORCE_GRAPH ? GRID_LAYOUT : FORCE_GRAPH));
-    $(FORCE_GRAPH_CLASS).click(() => setGraphLayout(FORCE_GRAPH));
-    $(GRID_LAYOUT_CLASS).click(() => setGraphLayout(GRID_LAYOUT));
+// Grid and Force Graph Layout
```
Comment (dondi) [inline web-client/public/js/setup-handlers.js]: The indentation on _this_ comment looks inconsistent with how the other comments have been re-indented. Pick a convention and stay with it.
```suggestion
    // Grid and Force Graph Layout
```

**Comment by mihirsamdarshi** [comment]
ready to be merged @dondi

**Comment by dondi** [comment]
@mihirsamdarshi There’s a failing test on this branch. Please fix it before we merge. (see Travis report for details)

File: web-client/public/js/setup-load-and-import-handlers.js [hunk 4]
```diff
@@ -51,6 +58,8 @@ const uploadHandler = (uploadRoute, uploader) => {
 };
 
 const networkErrorDisplayer = xhr => {
+    // re-enable upload button
+    disableUpload(false);
```
Comment (dondi) [inline web-client/public/js/setup-load-and-import-handlers.js]: Small naming note here: see how `disableUpload` as a function name doesn't line up with its parameter of true or false. ("disable upload to true" doesn't make a lot of plain English sense)

Instead, you can do this:
- `enableUpload()`
- `disableUpload()`
- `setUploadState(state)`

**Pull request merged**